### PR TITLE
feat: add host-aware navigation and job apply links

### PR DESCRIPTION
**Pull request by MarlonManabat**
## Summary
- factor header into reusable component with host-aware Post Job link
- add hostAware helper to build absolute URLs when an app host is set
- wire landing and job detail CTAs to use host-aware URLs and preserve return paths

## Testing
- `npm run lint` *(fails: next not found)*
- `npm ci` *(fails: unsupported engine / npm version)*
- `npx playwright test -c playwright.smoke.ts` *(fails: 403 Forbidden fetching playwright)*
- `npm test`
- `bash scripts/no-legacy.sh`

------
https://chatgpt.com/codex/tasks/task_e_68c830a1c8e48327b8ab4273e80d0041

### CODE REVIEW

File: src/app/browse-jobs/[id]/page.tsx [hunk 1]
```diff
@@ -1,29 +1,58 @@
...
-  } catch (err) {
-    console.error('failed to load job', err);
-    return <div className="text-gray-600">Job not found.</div>;
   }
+
+  // Build an auth-aware Apply link that preserves the return path after login
+  const returnTo = `/browse-jobs/${encodeURIComponent(String(job.id))}`;
+  const applyHref = job.applyUrl
+    ? hostAware(job.applyUrl)
+    : `/login?next=${encodeURIComponent(returnTo)}`;
```
[the remark this answers is not in the document]
Comment (MarlonManabat) [reply src/app/browse-jobs/[id]/page.tsx:41]: codex patch: gate /my-applications in middleware, make Apply CTA host-aware with preserved return path, and make Browse Jobs degrade gracefully when NEXT_PUBLIC_API_BASE_URL is missing (rendering test-friendly empty state with stable test IDs).
